apis/v1beta1/opentelemetrycollector_types.go

```diff
@@ -90,10 +90,10 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	TargetAllocator TargetAllocatorEmbedded `json:"targetAllocator,omitempty"`
 	// Mode represents how the collector should be deployed (deployment, daemonset, statefulset or sidecar)
-	// +optional
+	// +kubebuilder:validation:Enum=ModeDeployment;ModeDaemonSet;ModeStatefulSet;ModeSidecar
```

Comment:
This specific change is superfluous. If you look at the [generated CRD](https://github.com/open-telemetry/opentelemetry-operator/blob/91759b0138c6b58329b3420ef159688ff0070e4c/config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml#L2328), this validation is already there. If we look at what's missing compared to the [webhook](https://github.com/open-telemetry/opentelemetry-operator/blob/91759b0138c6b58329b3420ef159688ff0070e4c/apis/v1beta1/collector_webhook.go#L60), we should add a marker here to set the default to `deployment`.

Keep in mind that the markers deal with raw values of the field as they appear in the resource in K8s, not the aliases we have for them in Go code. So you should use `deployment` instead of `ModeDeployment`.